services/wfm/src/main/java/org/openkilda/wfm/topology/nbworker/services/ValidationService.java

```diff
@@ -0,0 +1,111 @@
+/* Copyright 2018 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.nbworker.services;
+
+import org.openkilda.messaging.info.rule.FlowEntry;
+import org.openkilda.messaging.info.rule.SwitchFlowEntries;
+import org.openkilda.messaging.info.switches.SyncRulesResponse;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import org.neo4j.driver.v1.Session;
+import org.neo4j.driver.v1.StatementResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ValidationService {
+    private static final Logger logger = LoggerFactory.getLogger(ValidationService.class);
+    public static final Set<Long> DEFAULT_RULES =
+            ImmutableSet.of(0x8000000000000001L, 0x8000000000000002L, 0x8000000000000003L);
+    private SwitchFlowEntries request;
+    private Session session;
+    private String switchId;
+
+    public ValidationService(SwitchFlowEntries request, Session session) {
+        this.request = request;
+        this.session = session;
+        switchId = request.getSwitchId().toString();
+    }
+
+    /**
+     * Runs validating process.
+     * @return validation result as RulesResponse
+     */
+    public SyncRulesResponse validate() {
+        logger.debug("Validating rules on switch {}", switchId);
+
+        Set<Long> expectedCookies = readSegmentsCookies();
+        expectedCookies.addAll(readIngressCookies());
+        logger.debug("Expected rules on switch {}: {}", switchId, expectedCookies);
+
+        Set<Long> presentedCookies = request.getFlowEntries().stream()
+                .map(FlowEntry::getCookie)
+                .collect(Collectors.toSet());
+        presentedCookies.removeAll(DEFAULT_RULES);
+        logger.debug("Presented rules on switch {}: {}", switchId, expectedCookies);
```

Comment:
I guess here should be 'presentedCookies' instead of 'expectedCookies'

Reply:
Oops. It's copy - paste problem )